plasmapy/diagnostics/proton_radiography.py

```diff
@@ -300,6 +305,212 @@ def _log(self, msg):
+            Raises a ValueError if the provided mesh location is not
+            between the source and the object grid.
+
+        """
+
+        location = _coerce_to_cartesian_si(location)
+        wire_radius = wire_diameter.si.value / 2
+
+        if not isinstance(extent, tuple):
+            extent = (extent,)
+
+        if len(extent) == 1:
+            radius = 0.5 * extent[0].si.value
+            width = extent[0].si.value
+            height = extent[0].si.value
+        elif len(extent) == 2:
+            radius = None
+            width = extent[0].si.value
+            height = extent[1].si.value
+        else:
+            raise ValueError(
+                "extent must be a tuple of 1 or 2 elements, but "
+                f"{len(extent)} elements were provided."
+            )
+
+        if not isinstance(nwires, tuple):
+            nwires = (nwires,)
+
+        if len(nwires) != 2:
+            nwires = (nwires[0], nwires[0])
+
+        # If no hdir/vdir is specified, calculate a default value
+        # If one is specified, make sure it is normalized
+        if mesh_hdir is None:
+            # Re-calculate the default here, in case the user
+            # specified a different det_hdir
+            mesh_hdir = self._default_detector_hdir()
+        else:
+            mesh_hdir = mesh_hdir / np.linalg.norm(mesh_hdir)
+
+        if mesh_vdir is None:
+            mesh_vdir = np.cross(mesh_hdir, self.det_n)
+            mesh_vdir = -mesh_vdir / np.linalg.norm(mesh_vdir)
+        else:
+            mesh_vdir = mesh_vdir / np.linalg.norm(mesh_vdir)
+
+        # Raise exception if mesh is AFTER the field grid
+        if np.linalg.norm(location - self.source) > np.linalg.norm(self.source):
+            raise ValueError(
+                f"The specified mesh location, {location},"
+                "is not between the source and the origin."
+            )
+
+        mesh_entry = {
+            "location": location,
+            "wire_radius": wire_radius,
+            "radius": radius,
+            "width": width,
+            "height": height,
+            "nwires": nwires,
+            "mesh_hdir": mesh_hdir,
+            "mesh_vdir": mesh_vdir,
+        }
+
+        self.mesh_list.append(mesh_entry)
+
+    def _apply_wire_mesh(
+        self,
+        location=None,
+        wire_radius=None,
+        radius=None,
+        width=None,
+        height=None,
+        nwires=None,
+        mesh_hdir=None,
+        mesh_vdir=None,
+    ):
+        """
+        Apply wire meshes that were added to self.mesh_list
+        """
+        x = self._coast_to_plane(location, mesh_hdir, mesh_vdir)
+
+        # Particle positions in 2D on the mesh plane
+        xloc = np.dot(x - location, mesh_hdir)
+        yloc = np.dot(x - location, mesh_vdir)
+
+        # Create an array in which 0 indicates that a particle has hit one of
+        # the wires and any other value indicates that it has not
+        no_hit = np.ones([self.nparticles])
```

Comment:
If nparticles is the shape you can drop the list, I think; also force a boolean dtype to conserve a little memory :)